src/scene/vcPOI.cpp

```diff
@@ -1000,6 +962,91 @@ vcRenderPolyInstance *vcPOI::AddNodeToRenderData(vcState *pProgramState, vcRende
   return pInstance;
 }
 
+void vcPOI::CalculateArea()
+{
+  m_area = 0.0;
+
+  if (m_line.numPoints < 3)
+    return;
+
+  //TODO This assumes we have a simple polygon.
+  //     Also we currently project the points onto the ground plane to find area,
+  //     we should be able to project these onto any plane.
+  m_area = udAbs(udSignedSimplePolygonArea(m_line.pPoints, (size_t)m_line.numPoints));
+}
+
+void vcPOI::CalculateTotalLength()
+{
+  m_totalLength = 0.0;
+
+  int j = udMax(0, m_line.numPoints - 1);
+  for (int i = 0; i < m_line.numPoints; i++)
+  {
+    double lineLength = udMag3(m_line.pPoints[j] - m_line.pPoints[i]);
+
+    if (m_line.closed || i > 0)
+      m_totalLength += lineLength;
+
+    j = i;
+  }
+}
+
+//TODO This is will work but not show the true centroid.
+// We should find the centroid of the convex hull.
+void vcPOI::CalculateCentroid()
+{
+  m_centroid = udDouble3::zero();
+
+  udDouble3 aabbMin = udDouble3::create(DBL_MAX, DBL_MAX, DBL_MAX);
+  udDouble3 aabbMax = udDouble3::create(DBL_MIN, DBL_MIN, DBL_MIN);
+
+  for (int p = 0; p < m_line.numPoints; p++)
+  {
+    for (int i = 0; i < 3; i++)
+    {
+      if (m_line.pPoints[p][i] < aabbMin[i])
+        aabbMin[i] = m_line.pPoints[p][i];
+      if (m_line.pPoints[p][i] > aabbMax[i])
+        aabbMax[i] = m_line.pPoints[p][i];
```

Comment:
I think you can just do;

`aabbMin = udMin(aabbMin, m_line.pPoints[p])`

Reply:
Acknowledged